File: LearningSystem.Core/Entities/HomeworkAssignment.cs
namespace LearningSystem.Core.Entities
{
    using System;
    using System.Collections.Generic;

    public class HomeworkAssignment : AuditInfo, IEntity
    {
        public string Content { get; set; }

        public int LectureId { get; set; }

        public Lecture Lecture { get; set; }
        
        public DateTime Deadline { get; set; }

        public List<HomeworkSubmission> HomeworkSubmissions { get; set; }
    }
}

File: LearningSystem.Web/ClientApp/src/app/_models/homeworkAssignment.ts
export class HomeworkAssignment {
  id: number;
  content: string;
  deadline: Date;
  hasUserSubmission: boolean;
}

export class HomeworkAssignmentInput {
  id: number;
  content: string;
  deadline: Date;
  lectureId: number;
  lectureName: string;
}
